bin/faucet/src/main.rs

```diff
@@ -72,6 +73,26 @@ async fn main() -> Result<(), FaucetError> {
 
             info!(target: COMPONENT, %config, "Initializing server");
 
+            let client_state_clone = Arc::clone(&faucet_state.client);
+
+            info!("Initializing chain tip updater");
+            thread::spawn(move || {
+                let sys = actix_web::rt::System::new();
+                sys.block_on(async move {
+                    loop {
+                        let state_clone_inner = Arc::clone(&client_state_clone);
+                        actix_web::rt::spawn(async move {
+                            let mut state = state_clone_inner.lock().await;
+                            state.update_current_block_number().await.unwrap();
+                        })
+                        .await
+                        .unwrap();
+
+                        thread::sleep(Duration::from_secs(CHAIN_TIP_UPDATER_INTERVAL));
```

Comment:
Feels weird using a normal sleep in async, shouldn't this be a tokio sleep? Or does it not matter because actix is single threaded?

Reply:
Yes, it should use a Tokio interval and a actix_web task should be enough too. Im changing it.